src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -0,0 +1,127 @@
+package org.broadinstitute.hellbender.engine.spark.datasources;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.genomics.utils.ReadUtils;
+import com.google.common.collect.Lists;
+import htsjdk.samtools.*;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.ReadsDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.GoogleGenomicsReadToGATKReadAdapter;
+import org.seqdoop.hadoop_bam.AnySAMInputFormat;
+import org.seqdoop.hadoop_bam.SAMRecordWritable;
+import org.seqdoop.hadoop_bam.util.SAMHeaderReader;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+/** Loads the reads from disk either serially (using samReaderFactory) or in parallel using Hadoop-BAM.
+ * The parallel code is a modified version of the example writing code from Hadoop-BAM.
+ */
+public class ReadsSparkSource {
+    private final JavaSparkContext ctx;
+    public ReadsSparkSource(JavaSparkContext ctx) {
+        this.ctx = ctx;
+    }
+
+    /**
+     * Loads Reads using samReaderFactory, then calling ctx.parallelize.
+     * @param bam file to load
+     * @return RDD of (SAMRecord-backed) GATKReads from the file.
+     */
+    public JavaRDD<GATKRead> getSerialReads(final String bam) {
```

Comment:
This method is only used by tests, so move there and rename getParallelReads to getReads.